lib/src/fields/form_builder_dropdown.dart

```diff
@@ -299,8 +300,28 @@ class _FormBuilderDropdownState<T>
   @override
   void didUpdateWidget(covariant FormBuilderDropdown<T> oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (widget.items != oldWidget.items) {
+
+    final oldValues = oldWidget.items.map((e) => e.value).toList();
+    final currentlyValues = widget.items.map((e) => e.value).toList();
+    final oldChilds = oldWidget.items.map((e) => e.child.toString()).toList();
+    final currentlyChilds =
+        widget.items.map((e) => e.child.toString()).toList();
+
+    if ((!listEquals(oldChilds, currentlyChilds) ||
+            !listEquals(oldValues, currentlyValues)) &&
+        (widget.items.contains(initialValue) ||
+            _emptyValidator<T>(initialValue))) {
       setValue(initialValue);
     }
   }
 }
+
+bool _emptyValidator<T>(T? value) {
```

Comment:
Could you put this function in a place where it can be reused? 

I feel it's pretty generic and could be used anywhere else if needed.

Maybe create a file named `lib/src/utils/utils.dart` and have it there? This is optional though.